pontoon/sync/vcs/repositories.py

```diff
@@ -71,15 +71,41 @@ def pull(self, source=None, target=None, branch=None):
 class PullFromHg(PullFromRepository):
 
     def pull(self, source=None, target=None):
-        log.debug("Mercurial: Update repository.")
-
         source = source or self.source
         target = target or self.target
 
-        # Undo local changes: Mercurial doesn't offer anything more elegant
-        command = ["rm", "-rf", target]
-        code, output, error = execute(command)
+        if os.path.exists(target):
+            log.debug("Mercurial: Update repository.")
+
+            # Make sure hg extensions needed to undo local changes are enabled
+            with open(os.path.join(target, '.hg/hgrc'), 'a+') as hgrc:
```

Comment:
I'd suggest to add a log info about what happens here because it may be later confusing to users if they'll notice unwanted changes in their hgrc file.